### ISSUE TITLE
Refresh Inbox Button

### ISSUE THREAD

**Issue by Appalbarry** [mailpile/Mailpile#1916]
I cannot find a way in the web interface to force refresh the Inbox - to pull down any new messages that may have arrived at Gmail.  There seems to be quite a lag between messages appearing in Gmail's inbox, and when they arrive in my Mailpile Inbox.

**Comment by Appalbarry** [comment]
By way of followup.  It seems that if you set a Gmail address to authenticate using OAuth it all works fine.  I had set one to use "password" and it seemed that it would start out fine and download existing mail, but never downloaded new messages.

**Comment by BjarniRunar** [comment]
I agree it would be nice to have a GUI element that lets the user force Mailpile to check for new mail "right now". This is not entirely trivial to do, but it would definitely be nice to have.

**Comment by Kabouik** [comment]
Is `rescan sources` the way to force it in command line, or is this a different thing?

**Comment by BjarniRunar** [comment]
@Kabouik Yes, `rescan sources` on the CLI should trigger a check for new mail. Refactoring that into an API method which can be invoked from the web is the most straightforward way to solve this issue; the complications mostly have to do with providing good feedback once the button has been pressed.

**Comment by Kabouik** [comment]
Thanks @BjarniRunar. Wanted to make sure I was doing the right thing and not something totally irrelevant when I was typing it in the terminal to refresh the inbox. :>

**Comment by spinus** [comment]
it's very very needed! Even clicking on folder name could trigger the action so UI look same (IMHO)

**Comment by halfshavedyaks** [comment]
Related issue - how do I set the frequency or conditions with which mailpile checks for new mail? I'm used to setting each account to only check for new mail when I manually ask it to. This is a reversal for me - opaque automatic checking and no option to check manually. 

I'm willing to change how I do things  - but I'd like to understand the default in mailpile as a start.

**Comment by BjarniRunar** [comment]
It's a bit complicated; each mail source has its own schedule, and they all try to be "clever" and notice new mail as promptly as possible (with IMAP that means using the IDLE extension, with local files that means frequently checking modification times).

This wiki page describes how Mail sources are structured and configured: https://github.com/mailpile/Mailpile/wiki/Mail-sources - at the Mailpile CLI, you can exploe and manipulate settings with the `print` and `set` commands. A good starting point would be `print -flat sources`.

The system was not designed with completely manual checking in mind and there's no straightforward way to configure that today. It's an interesting idea, if we did have a "check mail" button, it would make sense to add that as an option.

**Comment by halfshavedyaks** [comment]
Looks like I'll just live with that for now. I tend to prefer simpleminded settings where possible. 

I think any schedule of checks should be accessible in the gui really and that manual checking only should be an option - but maybe I'm set in my simpleminded ways.

I'm also looking for an option to set when messages are deleted from the server.

print sources shows my account as IMAP but it was originally set up as pop3 on thunderbird... I'm not sure if that matters.

**Comment by spinus** [comment]
Is this issue addressed other way by any chance? Do anybody know a workaround to fetch/refresh emails?
For me this is really a blocker to use the software as it gives me wrong impression where there is no new email, but there is on the server.

**Comment by BjarniRunar** [comment]
@spinus As mentioned above, in the Mailpile CLI you can type `rescan sources` to trigger a check ASAP.

If you have keyboard shortcuts enabled in the web UI (see Settings and Tools > Preferences), this is as simple as pressing 't' and then typing `rescan sources` and then hitting ESC to dismiss the web terminal. You can also click your way there: Settings and Tools > CLI.

**Comment by spinus** [comment]
@BjarniRunar I appreciate this way.

>  this is as simple as pressing 't' and then typing rescan sources and then hitting ESC to dismiss the web terminal.

I wonder why gmail, yahoo or outlook didn't figure this out, so easy. Should we let them know!?

Sorry for little sarcasm, but this thread also feels like that. That's only my opinion, but IMHO that function is no-brainer for adoption of the project. I wonder how many people thought this app is not working and just stopped looking at it more.

Is it expected behaviour or if someone create PR to add this function you consider merging?
